nakadi-java-client/src/main/java/nakadi/EventResource.java

```diff
@@ -28,6 +28,18 @@
   EventResource retryPolicy(RetryPolicy retryPolicy);
 
   /**
+   * Set the flow ID that should be included in the header of all future requests.
+   * <p>
+   * The default value is null. If the current value is null, a flow ID will be generated by
+   * the client.
+   * </p>
+   *
+   * @param flowId the flow ID.
+   * @return this
+   */
+  EventResource flowId(String flowId);
```

Comment:
(Sorry for the delay, for some reason I didn't get notifications for this, or missed them)

Nice, looks like what's here will work. 

I'd love this to be a property/value option, but then again flowId is hardcoded into the server API as well. So maybe this is fine and a rebase to get it merged is good enough.